src/htm/algorithms/Connections.cpp

```diff
@@ -645,6 +652,10 @@ std::ostream& operator<< (std::ostream& stream, const Connections& self)
 
   stream << "    Synapses Dead (" << (Real) synapsesDead / self.numSynapses()
          << "%) Saturated (" <<   (Real) synapsesSaturated / self.numSynapses() << "%)" << std::endl;
+  stream << "    Synapses pruned (" << (Real) self.prunedSyns_ / self.numSynapses() 
+	 << "%) Segments pruned (" << (Real) self.prunedSegs_ / self.numSegments() << "%)" << std::endl;
+  stream << "    Buffer for destroyed synapses: " << self.destroyedSynapses_.size() << " \t buffer for destr. segments: "
+	 << self.destroyedSegments_.size() << std::endl; 
```

Comment:
can you help me make "not completely artificial" test where synapses & segments are heavily pruned? 
I'd like to:
- validate the code is actually (enough) used
- verify the `destroyed*` buffers don't grow too much, cap them if needed. 

This is what I get on Hotgym -> not really used at all. 
```
TM Connections:
Inputs (4653) ~> Outputs (16384) via Segments (32626)
Segments on Cell Min/Mean/Max 0 / 1.99133 / 40
Potential Synapses on Segment Min/Mean/Max 10 / 10.5655 / 49
Connected Synapses on Segment Min/Mean/Max 0 / 0.356985 / 34
Synapses Dead (0%) Saturated (0.0110586%)
Synapses pruned (0.0708685%) Segments pruned (0%)
Buffer for destroyed synapses: 0     buffer for destr. segments: 0
```

Reply:
You should be able to make it prune synapses by changing the data set.  For example train it on sine waves and then move to saw tooth waves.

I think it's normal and good that it does not prune very many synapses / segments.

Segments should really only be pruned when the HTM reaches capacity.

Reply:
> You should be able to make it prune synapses by changing the data set. For example train it on sine waves and then move to saw tooth waves.

would be nice to have some test on "switching contexts", where we observe some more synapse pruning shortly after the change.

Reply:
#553